File: internal/model/task.go
package model

import (
	"database/sql"

	"github.com/library-squirrel/wails/pkg/model"
)

// Task 任务
type Task struct {
	*model.BaseEntity                   // 嵌入基础实体
	IsCollection         sql.NullInt64  `gorm:"column:is_collection" json:"isCollection"`
	Pid                  sql.NullInt64  `gorm:"column:pid" json:"pid"`
	TaskName             sql.NullString `gorm:"column:task_name" json:"taskName"`
	SiteID               sql.NullInt64  `gorm:"column:site_id" json:"siteId"`
	SiteWorkID           sql.NullString `gorm:"column:site_work_id" json:"siteWorkId"`
	URL                  sql.NullString `gorm:"column:url" json:"url"`
	Status               int            `gorm:"column:status" json:"status"`
	PendingResourceID    sql.NullInt64  `gorm:"column:pending_resource_id" json:"pendingResourceId"`
	Continuable          sql.NullInt64  `gorm:"column:continuable" json:"continuable"`
	PluginPublicID       sql.NullString `gorm:"column:plugin_public_id" json:"pluginPublicId"`
	PluginContributionID sql.NullString `gorm:"column:plugin_contribution_id" json:"pluginContributionId"`
	PluginData           sql.NullString `gorm:"column:plugin_data" json:"pluginData"`
	ErrorMessage         sql.NullString `gorm:"column:error_message" json:"errorMessage"`
}

// NewTask 创建任务
func NewTask() *Task {
	return &Task{
		BaseEntity: &model.BaseEntity{},
	}
}

// TableName 指定表名
func (Task) TableName() string {
	return "task"
}

File: frontend/src/model/model/entity/Task.ts
import BaseEntity from '../base/BaseEntity.ts'
import { notNullish } from '@renderer/utils/CommonUtil.ts'

/**
 * 任务
 */
export default class Task extends BaseEntity {
  /**
   * 是否是父任务
   */
  isCollection: boolean | undefined | null

  /**
   * 上级任务id
   */
  pid: number | undefined | null

  /**
   * 任务名称
   */
  taskName: string | undefined | null

  /**
   * 任务的站点id
   */
  siteId: number | undefined | null

  /**
   * 站点作品id
   */
  siteWorkId: string | undefined | null

  /**
   * 当任务是父任务时，url存储创建此父任务所使用的url，否则存储的是下载资源的链接（由于存在临时的下载链接，此字段可能没有作用）
   */
  url: string | undefined | null

  /**
   * 状态（0：未开始，1：进行中，2：暂停，3：已完成，4：失败）
   */
  status: number | undefined | null

  /**
   * 保存中的资源id
   */
  pendingResourceId: number | undefined | null

  /**
   * 资源是否支持续传
   */
  continuable: boolean | undefined | null

  /**
   * 插件公开id
   */
  pluginPublicId: string | undefined | null

  /**
   * 插件贡献点id
   */
  pluginContributionId: string | undefined | null

  /**
   * 插件数据
   */
  pluginData: { [key: string]: unknown } | string | undefined | null

  /**
   * 错误信息
   */
  errorMessage: string | undefined | null

  constructor(task?: Task) {
    super(task)
    if (notNullish(task)) {
      this.id = task.id
      this.isCollection = task.isCollection
      this.pid = task.pid
      this.taskName = task.taskName
      this.siteId = task.siteId
      this.siteWorkId = task.siteWorkId
      this.url = task.url
      this.status = task.status
      this.pendingResourceId = task.pendingResourceId
      this.continuable = task.continuable
      this.pluginPublicId = task.pluginPublicId
      this.pluginContributionId = task.pluginContributionId
      if (typeof task.pluginData === 'string') {
        this.pluginData = JSON.parse(task.pluginData)
      } else {
        this.pluginData = task.pluginData
      }
      this.errorMessage = task.errorMessage
    }
  }
}
